src/typescript/frontend/src/components/modals/BaseModal.tsx

```diff
@@ -23,6 +23,8 @@ export const BaseModal: React.FC<
   children,
   className,
 }) => {
+  const [hoveringOnCloseButton, setHoveringOnCloseButton] =
```

Comment:
May I ask why we have to use the useState hook to store hovering state?

Reply:
because i'm unable to produce the same effect using pure css, so we need to track hover state to trigger blue outline